pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -33,22 +33,133 @@ import (
 	"github.com/golang/glog"
 )
 
-// Cache saves and reuses the output of predicate functions. Use RunPredicate to
-// get or update the cached results. An appropriate Invalidate* function should
-// be called when some predicate results are no longer valid.
+// nodeMap stores a *Cache for each node.
+type nodeMap map[string]*NodeCache
+
+// Cache is a thread safe map saves and reuses the output of predicate functions,
+// it uses node name as key to access those cached results.
 //
-// Internally, results are keyed by node name, predicate name, and "equivalence
+// Internally, results are keyed by predicate name, and "equivalence
 // class". (Equivalence class is defined in the `Class` type.) Saved results
 // will be reused until an appropriate invalidation function is called.
 type Cache struct {
-	mu    sync.RWMutex
-	cache nodeMap
+	// NOTE(harry): Theoretically sync.Map has better performance in machine with 8+ CPUs, while
+	// the reality is lock contention in first level cache is rare.
+	mu          sync.RWMutex
+	nodeToCache nodeMap
 }
 
-// NewCache returns an empty Cache.
+// NewCache create an empty equiv class cache.
 func NewCache() *Cache {
 	return &Cache{
-		cache: make(nodeMap),
+		nodeToCache: make(nodeMap),
+	}
+}
+
+// ForNode returns the existing NodeCache for given node if present. Otherwise,
+// it creates the NodeCache and returns it.
+// The boolean flag is true if the value was loaded, false if created.
+func (c *Cache) ForNode(name string) (nodeCache *NodeCache, exists bool) {
```

Comment:
I'd prefer something like GetNodeCache instead of ForNode.

Reply:
Haha, sure, that's the the original version.